pkg/kubelet/cm/dra/manager_test.go

```diff
@@ -59,9 +69,14 @@ func (s *fakeDRADriverGRPCServer) NodePrepareResources(ctx context.Context, req
 	if s.timeout != nil {
 		time.Sleep(*s.timeout)
 	}
-	deviceName := "claim-" + req.Claims[0].Uid
-	result := s.driverName + "/" + driverClassName + "=" + deviceName
-	return &drapbv1.NodePrepareResourcesResponse{Claims: map[string]*drapbv1.NodePrepareResourceResponse{req.Claims[0].Uid: {CDIDevices: []string{result}}}}, nil
+
+	resp := s.prepareResourcesResponse
+	if resp == nil {
+		deviceName := "claim-" + req.Claims[0].Uid
+		result := s.driverName + "/" + driverClassName + "=" + deviceName
+		resp = &drapbv1.NodePrepareResourcesResponse{Claims: map[string]*drapbv1.NodePrepareResourceResponse{req.Claims[0].Uid: {CDIDevices: []string{result}}}}
+	}
```

Comment:
@bitoku I'm not sure I understand this logic. Why do we store and use response of the previous call? Can you elaborate?

Reply:
got it. Instead of setting it explicitly can we just pass expected response to the `setupFakeDRADriverGRPCServer` as a parameter?

Reply:
My concern is inconsistency. So far we passed all required parameters to the `setupFakeDRADriverGRPCServer` and it worked just fine. Now we have two new setters and I don't see the reason. As both fields are pointers we can pass nil if one is not needed. 

> And if we move this logic to other place, we can no longer return the responsive value to the request and every test cases have to have the return value of the function.

I din't get this part, sorry. Can you elaborate?

Reply:
> I din't get this part, sorry. Can you elaborate?

Never mind.
I was worried it can't return the response according to the input `req`, but if we give the expected return value as an input, we don't have to care about it.

Fair enough. I will fix that. thank you!